lib/game.rb

```diff
@@ -0,0 +1,38 @@
+class Game
+  def initialize(rolls)
+    @rolls = rolls
+  end
+
+  def score
+    score = 0
+    @rolls.each_with_index do |frame, i|
+      score += frame_score(frame, i)
+    end
+    score
+  end
+
+  private
+
+  def frame_score(frame, index)
+    return 0 if index > 9                                    # don't count the bonus on their own
+    frame_score = frame.inject(0,:+)
+    frame_score += strike_or_spare(frame, index) if frame_score == 10 # if a strike or spare
```

Comment:
Could integrate the if statement into the strike_or_space method. Benefit of this is that strike_or_spare could be applied to all frames without generating erroneous bonuses. See my comment below about a guard for non-10-scoring frames in the strike_or_space method